packages/external-db-authorization/lib/services/authentication_service.js

```diff
@@ -0,0 +1,73 @@
+const passport = require('passport')
+const { Router } = require('express')
+
+class AuthenticationService {
+    constructor({ authProvider, authInformation }) {
+        this.authProvider = authProvider
+        this.authInformation = authInformation
+
+        this.authName = 'external-db-authentication'
+        
+        this.loginPath = '/auth/login'
+        this.loginCallbackPath = '/auth/callback'
+        this.logoutPath = '/auth/logout'
+
+        this.authRouter = new Router()
+    }
+
+    router() {
+        this.authRouter.use(passport.initialize())
+        this.authRouter.use(passport.session())
+        
+        passport.serializeUser((user, done) => done(null, user))
+        passport.deserializeUser((user, done) => done(null, user))
+        passport.use(this.authName, this.authProvider)
+
+        this.createLoginHandler()
+        this.createLoginCallbackHandler()
+        this.createLogoutHandler()
+        
+        this.createAuthenticationMiddleware()
+        this.createAuthErrorMiddleware()
+        
+        return this.authRouter
+    }
+
+    createLoginHandler() {
+
+        this.authRouter.use(this.loginPath, passport.authenticate(this.authName))
+    }
+
+    createLoginCallbackHandler() {
+        const redirectOptions = { successReturnToOrRedirect: '/', failureRedirect: '/auth/login' }
```

Comment:
I'm not sure that it supposed to be in the constructor (maybe for the tests you tell me), 
but have `this.loginPath` in constructor, maybe use it here in `failureRedirect` instead hard-coded?